src/Platform/Microsoft.Testing.Platform/OutputDevice/Console/ConsoleLogger.cs

```diff
@@ -0,0 +1,702 @@
+    {
+        if (RoslynString.IsNullOrWhiteSpace(message))
+        {
+            return;
+        }
+
+        if (!message.Contains('\n'))
+        {
+            terminal.Append(indent);
+            terminal.Append(message.TrimStart(' '));
+            return;
+        }
+
+        string[] lines = message.Split(NewLineStrings, StringSplitOptions.None);
+        foreach (string line in lines)
+        {
+            string trimmedLine = line.TrimStart(' ');
+            // Here we could check if the messages are longer than then line, and reflow them so a long line is split into multiple
+            // and prepended by the respective indentation.
+            // But this does not play nicely with ANSI escape codes. And if you
+            // run in narrow terminal and then widen it the text does not reflow correctly. And you also have harder time copying
+            // values when the assertion message is longer.
+            terminal.Append(indent);
+            terminal.Append(trimmedLine);
+            terminal.AppendLine();
+        }
+    }
+
+    internal void AssemblyRunCompleted(string assembly, string? targetFramework, string? architecture)
+    {
+        TestModule assemblyRun = GetOrAddAssemblyRun(assembly, targetFramework, architecture);
+        assemblyRun.Stopwatch.Stop();
+
+        _consoleWithProgress.RemoveWorker(assemblyRun.SlotIndex);
+
+        if (_options.ShowAssembly && _options.ShowAssemblyStartAndComplete)
+        {
+            _consoleWithProgress.WriteToTerminal(terminal => AppendAssemblySummary(assemblyRun, terminal));
+        }
+    }
+
+    private static void AppendAssemblySummary(TestModule assemblyRun, ITerminal terminal)
+    {
+        int failedTests = assemblyRun.FailedTests + assemblyRun.CancelledTests + assemblyRun.TimedOutTests;
+        int warnings = 0;
+
+        AppendAssemblyLinkTargetFrameworkAndArchitecture(terminal, assemblyRun.Assembly, assemblyRun.TargetFramework, assemblyRun.Architecture);
+        terminal.Append(' ');
+        AppendAssemblyResult(terminal, assemblyRun.FailedTests == 0, failedTests, warnings);
+        terminal.Append(' ');
+        AppendDuration(terminal, assemblyRun.Stopwatch.Elapsed);
+    }
+
+    private static void AppendDuration(ITerminal terminal, TimeSpan duration, bool wrapInParentheses = true, bool colorize = true)
+    {
+        if (colorize)
+        {
+            terminal.SetColor(TerminalColor.Gray);
+        }
+
+        if (wrapInParentheses)
+        {
+            terminal.Append('(');
+        }
+
+        if (duration.TotalMilliseconds < 1000)
+        {
+            terminal.Append(duration.TotalMilliseconds.ToString("F0", CultureInfo.CurrentCulture));
+            terminal.Append("ms");
+        }
+        else
+        {
+            terminal.Append(duration.TotalSeconds.ToString("F1", CultureInfo.CurrentCulture));
+            terminal.Append('s');
```

Comment:
What about minutes and hours? We know it's possible to have such "slow" tests

Reply:
I used the standard human readable rendering we use in other places.